testing/internal/e2e/docker/docker.go

```diff
@@ -0,0 +1,236 @@
+	require.NoError(t, err)
+	_, err = f.WriteString(string(data))
+	require.NoError(t, err)
+
+	networkAlias := "e2epostgres"
+	postgresDb := "e2eboundarydb"
+	postgresUser := "e2eboundary"
+	postgresPassword := "e2eboundary"
+	req := testcontainers.ContainerRequest{
+		Image: "postgres:13-alpine",
+		Cmd:   []string{"postgres", "-c", "config_file=/etc/postgresql/postgresql.conf"},
+		Env: map[string]string{
+			"POSTGRES_DB":       postgresDb,
+			"POSTGRES_USER":     postgresUser,
+			"POSTGRES_PASSWORD": postgresPassword,
+		},
+		Mounts: testcontainers.ContainerMounts{{
+			Source:   testcontainers.GenericBindMountSource{HostPath: path.Dir(postgresConfigFilePath)},
+			Target:   "/etc/postgresql/",
+			ReadOnly: true,
+		}},
+		Networks:       []string{networkName},
+		NetworkAliases: map[string][]string{networkName: {networkAlias}},
+		ExposedPorts:   []string{"5432/tcp"},
+		WaitingFor:     wait.ForLog("database system is ready to accept connections"),
+	}
+
+	ctx := context.Background()
+	postgresC, err := testcontainers.GenericContainer(ctx, testcontainers.GenericContainerRequest{
+		ContainerRequest: req,
+		Started:          true,
+	})
+	require.NoError(t, err)
+
+	uriNetwork := fmt.Sprintf("postgres://%s:%s@%v:%v/%s?sslmode=disable",
+		postgresUser,
+		postgresPassword,
+		networkAlias,
+		"5432",
+		postgresDb,
+	)
+
+	mappedPort, err := postgresC.MappedPort(ctx, "5432")
+	require.NoError(t, err)
+	uriLocalhost := fmt.Sprintf("postgres://%s:%s@%v:%v/%s?sslmode=disable",
+		postgresUser,
+		postgresPassword,
+		"localhost",
+		mappedPort.Port(),
+		postgresDb,
+	)
+
+	return &e2eContainer{
+		Container:    postgresC,
+		UriNetwork:   uriNetwork,
+		UriLocalhost: uriLocalhost,
+	}
+}
+
+// InitBoundaryDatabase starts a boundary container (of the latest released version) and initializes a
+// postgres database (using `boundary database init`) at the specified postgres URI.
+// Returns information about the container
+func InitBoundaryDatabase(t testing.TB, postgresURI string, networkName string) *e2eContainer {
+	t.Log("Initializing postgres database...")
+
+	data, err := ControllerConfigFile.ReadFile("testdata/boundary-config.hcl")
+	require.NoError(t, err)
+	boundaryConfigFilePath := fmt.Sprintf("%s/%s", t.TempDir(), "config.hcl")
+	f, err := os.Create(boundaryConfigFilePath)
+	require.NoError(t, err)
+	_, err = f.WriteString(string(data))
+	require.NoError(t, err)
+
+	req := testcontainers.ContainerRequest{
+		Image: "hashicorp/boundary:latest",
+		Cmd:   []string{"boundary", "database", "init", "-config", "/boundary/config.hcl", "-format", "json"},
+		Env: map[string]string{
+			"BOUNDARY_POSTGRES_URL": postgresURI,
+			"SKIP_CHOWN":            "true",
+		},
+		AlwaysPullImage: true, // ensures the latest release is used
+		CapAdd:          []string{"IPC_LOCK"},
+		Mounts: testcontainers.ContainerMounts{{
+			Source:   testcontainers.GenericBindMountSource{HostPath: path.Dir(boundaryConfigFilePath)},
```

Comment:
Instead of adding mounts, why not just copy the file? https://golang.testcontainers.org/features/copy_file/. Also, we can just ignore the `embed.FS` and copy directly from `testdata`. I swear we talked about this sort of thing before, did I change my mind again? 🤔

Reply:
Done. I think I was missing something when I was trying it out earlier. Figured something out now.